### PR TITLE
Feature/40314

### PR DESCRIPTION
**Pull request by Firoz-627**
Please go through these swagger files and review it

### CODE REVIEW

File: src/repositories/versionable/VersionableRepository.ts [hunk 1]
```diff
@@ -54,8 +54,8 @@ export default class VersionableRepository<D extends mongoose.Document, M extend
     return await this.updateData(deletedById, id);
   }
 
-  public async list(sortBy, userRole, skip, limit) {
-    return this.modelType.find({ role: userRole, deletedAt: undefined }).sort(sortBy).skip(Number(skip)).limit(Number(limit));
+  public async list (userRole, skip, limit, sortBy, searchBy) {
```
Comment (suraj1249) [inline src/repositories/versionable/VersionableRepository.ts]: Add UserRole  to search by

File: src/controllers/trainee/Controllers.ts [hunk 2]
```diff
@@ -80,7 +81,13 @@ class TraineeController {
       else {
         sortBy = { updatedAt: 1 };
       }
-      const user = await UserRepository.list(sortBy, 'trainee', req.query.skip, req.query.limit);
+      if (req.query.search !== undefined ) {
+        user = await UserRepository.list('trainee', req.query.skip, req.query.limit, sortBy, {name: { $regex: req.query.search.toLowerCase()}});
+        const List = await UserRepository.list('trainee', req.query.skip, req.query.limit, sortBy, {email: { $regex: req.query.search.toLowerCase()}});
```
Comment (suraj1249) [inline src/controllers/trainee/Controllers.ts]: No need to run query multiple times. modify your search object to do it at once

File: src/repositories/versionable/VersionableRepository.ts (hunk 1 above)
Comment (Firoz-627) [reply src/repositories/versionable/VersionableRepository.ts]: resolved

File: src/controllers/trainee/Controllers.ts (hunk 2 above)
Comment (Firoz-627) [reply src/controllers/trainee/Controllers.ts]: resolved

**Pull request merged**